src/components/tl/mlx5/mcast/tl_mlx5_mcast_team.c

```diff
@@ -118,26 +118,11 @@ ucc_status_t ucc_tl_mlx5_mcast_team_init(ucc_base_context_t *base_context,
 
     memcpy(&comm->params, conf_params, sizeof(*conf_params));
 
-    comm->wsize     = conf_params->wsize;
-    comm->max_eager = conf_params->max_eager;
-    comm->comm_id   = team_params->id;
-    comm->ctx       = mcast_context;
-    comm->grh_buf   = (char *)ucc_malloc(GRH_LENGTH * sizeof(char), "grh_buf");
-    if (!comm->grh_buf) {
-        status = UCC_ERR_NO_MEMORY;
-        goto cleanup;
-    }
-
-    memset(comm->grh_buf, 0, GRH_LENGTH);
-    
-    comm->grh_mr = ibv_reg_mr(mcast_context->pd, comm->grh_buf, GRH_LENGTH,
-                              IBV_ACCESS_REMOTE_WRITE |
-                              IBV_ACCESS_LOCAL_WRITE);
-    if (!comm->grh_mr) {
-        tl_error(mcast_context->lib, "could not register memory for GRH, errno %d", errno);
-        status = UCC_ERR_NO_RESOURCE;
-        goto cleanup;
-    }
+    comm->wsize              = conf_params->wsize;
```

Comment:
Can you explain why these manual copies are needed in addition to the `memcpy` above?

Reply:
The manual copy above this line just keeps a copy of the configuration parameters in the ```comm->params``` var.  
On the other hand, the values that get used during runtime are read from the variables defined in ```ucc_tl_mlx5_mcast_coll_comm_t``` (like ```comm->wsize```).  Such variables are not necessarily the same as the config params.

Reply:
but why do we need to copy twice the same data like `conf_params->wsize`?

Reply:
I am copying it once into comm->wsize, can you please clarify?

Reply:
Yes, there are two copies, but the destination buffers are different. comm->param is a cons value but comm->wsize is not constant and could change later.